Signal/src/ViewControllers/HomeView/HomeViewCell.m

```diff
@@ -0,0 +1,424 @@
+                                                    initWithString:displayableText
+                                                        attributes:@{
+                                                            NSFontAttributeName :
+                                                                (hasUnreadMessages ? self.snippetFont.ows_medium
+                                                                                   : self.snippetFont),
+                                                            NSForegroundColorAttributeName :
+                                                                (hasUnreadMessages ? [UIColor ows_blackColor]
+                                                                                   : [UIColor lightGrayColor]),
+                                                        }]];
+        }
+    }
+
+    return snippetText;
+}
+
+#pragma mark - Date formatting
+
+- (NSAttributedString *)attributedStringForDate:(nullable NSDate *)date
+{
+    if (date == nil) {
+        OWSProdLogAndFail(@"%@ date was unexpectedly nil", self.logTag);
+        return [NSAttributedString new];
+    }
+
+    NSString *dateTimeString;
+    if (![DateUtil dateIsThisYear:date]) {
+        dateTimeString = [[DateUtil dateFormatter] stringFromDate:date];
+    } else if ([DateUtil dateIsOlderThanOneWeek:date]) {
+        dateTimeString = [[DateUtil monthAndDayFormatter] stringFromDate:date];
+    } else if ([DateUtil dateIsOlderThanToday:date]) {
+        dateTimeString = [[DateUtil shortDayOfWeekFormatter] stringFromDate:date];
+    } else {
+        dateTimeString = [[DateUtil timeFormatter] stringFromDate:date];
+    }
+
+    return [[NSAttributedString alloc] initWithString:dateTimeString
+                                           attributes:@{
+                                               NSForegroundColorAttributeName : [UIColor ows_darkGrayColor],
+                                               NSFontAttributeName : self.dateTimeFont,
+                                           }];
+}
+
+#pragma mark - Constants
+
+- (UIFont *)dateTimeFont
+{
+    return [UIFont ows_dynamicTypeFootnoteFont];
+}
+
+- (UIFont *)snippetFont
+{
+    return [UIFont ows_dynamicTypeBodyFont];
+}
+
+- (UIFont *)nameFont
+{
+    return [UIFont ows_dynamicTypeBodyFont].ows_medium;
+}
+
+// Used for profile names.
+- (UIFont *)nameSecondaryFont
+{
+    return [UIFont ows_dynamicTypeFootnoteFont];
+}
+
+#pragma mark - Reuse
+
+- (void)prepareForReuse
+{
+    [super prepareForReuse];
+
+    [NSLayoutConstraint deactivateConstraints:self.viewConstraints];
+    [self.viewConstraints removeAllObjects];
+
+    self.thread = nil;
+    self.contactsManager = nil;
+
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+#pragma mark - Name
+
+- (void)otherUsersProfileDidChange:(NSNotification *)notification
+{
+    OWSAssertIsOnMainThread();
+
+    NSString *recipientId = notification.userInfo[kNSNotificationKey_ProfileRecipientId];
```

Comment:
Can you make sure this code path is in the test plan - seems like an easy one to not notice if broken.

Reply:
Done.